### PR TITLE
Feature/map focus button

### PR DESCRIPTION
**Pull request by lucieho**
This feature adds the possibility to zoom out when seeing a path (yesterday/before yesterday) on the map. You click on the path you want to see, this makes the camera of the map focus on the first coordinates of the path. A new small button (square) appears on the top left: if you click on it, the map's camera adjusts itself so that you can see the path (the one you just clicked on) in its entirety.

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sdp/map/fragment/MapActivityTest.java [hunk 1]
```diff
@@ -256,22 +248,87 @@ public void cameraTargetsPathWhenToggle() throws Throwable {
...
 
-        yesterdayInfectedLayerIsSetWhenNotEmpty();
         if (infectedCoordIsEmpty) {
             infectedLayerVisibilityChangesWhenNotEmpty();
         }
 
         PathsHandler.TEST_NON_EMPTY_LIST = false;
     }
 
+    @Test
```
[the remark this answers is not in the document]
Comment (lucieho) [reply app/src/androidTest/java/ch/epfl/sdp/map/fragment/MapActivityTest.java:268]: man it's just one line

File: app/src/main/java/ch/epfl/sdp/map/PathsHandler.java [hunk 2]
```diff
@@ -193,6 +196,32 @@ public void setCameraPosition(int day) {
...
+            map.easeCamera(CameraUpdateFactory.newLatLngBounds(latLngBounds, 100), 2000);
+        }
+    }
+
+    private LatLngBounds setLatLngBounds(int day) {
+        List<Point> pathCoord = day == R.string.yesterday ? yesterdayPathCoordinates :
+                beforeYesterdayPathCoordinates;
+        Log.d("SETLATLNGBNDS: ", "pathCoord is empty? " + pathCoord.isEmpty());
+        LatLngBounds.Builder boundsBuilder = new LatLngBounds.Builder();
+        double lat;
```
Comment (Ulydev) [inline app/src/main/java/ch/epfl/sdp/map/PathsHandler.java]: Is `lat` and `lng` relevant outside of the scope of the for?

**Review by Ulydev — APPROVED** [review]
Great job!

File: app/src/main/java/ch/epfl/sdp/map/PathsHandler.java (hunk 2 above)
Comment (lucieho) [reply app/src/main/java/ch/epfl/sdp/map/PathsHandler.java]: Thanks for this remark, I can put them inside the for

File: app/src/main/java/ch/epfl/sdp/map/fragment/MapFragment.java [hunk 3]
```diff
@@ -363,15 +382,12 @@ public void onRFACItemLabelClick(int position, RFACLabelItem item) {
     public void onRFACItemIconClick(int position, RFACLabelItem item) {
         String day = position == 0 ? getString(R.string.yesterday) : getString(R.string.before_yesterday);
```
Comment (TheBlueHawk) [inline app/src/main/java/ch/epfl/sdp/map/fragment/MapFragment.java]: This variable is never used

**Review by TheBlueHawk — COMMENTED** [review]
Good job, just a general remark: would have not been better to use a package private variable for yesterday and before-yesterday instead of using the R.String resources?

File: app/src/main/java/ch/epfl/sdp/map/fragment/MapFragment.java (hunk 3 above)
Comment (lucieho) [reply app/src/main/java/ch/epfl/sdp/map/fragment/MapFragment.java]: You're right, it's not useful anymore, I missed this, thanks!

**Review by TheBlueHawk — APPROVED** [review]

**Pull request merged**